torch/_inductor/scheduler.py

```diff
@@ -810,6 +849,11 @@ def fuse_nodes_once(self):
                 )
         self.nodes = sorted(fused_nodes, key=lambda x: x.min_order)
         self.topological_sort_schedule()
+        self.prune_redundant_deps()
```

Comment:
Let's move this after all the fusions passes have finished (fuse_nodes_once is called multiple times).

Reply:
Discussed offline: this needs to occur as we try fusing in order to enable more fusions